generator_gomod.go

```diff
@@ -30,20 +32,27 @@ func (s *Spec) HasGomods() bool {
 
 func withGomod(g *SourceGenerator, srcSt, worker llb.State, opts ...llb.ConstraintsOpt) func(llb.State) llb.State {
 	return func(in llb.State) llb.State {
-		const workDir = "/work/src"
-		var srcMount llb.RunOption
-		if g.Subpath != "" {
-			srcMount = llb.AddMount(workDir, srcSt, llb.SourcePath(g.Subpath))
-		} else {
-			srcMount = llb.AddMount(workDir, srcSt)
+		workDir := "/work/src"
+		joinedWorkDir := filepath.Join(workDir, g.Subpath)
+		srcMount := llb.AddMount(workDir, srcSt)
+
+		paths := g.Gomod.Paths
+		if g.Gomod.Paths == nil {
+			paths = []string{"."}
+		}
+		states := make([]llb.State, 0, len(paths))
```

Comment:
Since there can be duplicate modules that need to get downloaded, instead of using multiple states and merging them, can we build up a single state?